p2p/module.go

```diff
@@ -0,0 +1,193 @@
+package p2p
+
+import (
+	"errors"
+	"log"
+	"net"
+	"pocket/p2p/types"
+	"pocket/shared/config"
+	"pocket/shared/modules"
+	"pocket/shared/types"
+	"sync"
+
+	"go.uber.org/atomic"
+	"google.golang.org/protobuf/types/known/anypb"
+)
+
+var (
+	MaxInbound           uint = 128
+	MaxOutbound          uint = 128
+	WireByteHeaderLength int  = 9
+	ReadBufferSize       int  = (1024 * 4)
+	WriteBufferSize      int  = (1024 * 4)
+	ReadDeadlineMs       int  = 400
+)
+
+var (
+	Protocol = "tcp"
+	Address  = "localhost:30303"
+)
+
+type P2PModule struct {
+	modules.NetworkModule
+
+	bus    modules.Bus
+	config *config.P2PConfig
+	c      *dcodec
+
+	id           uint64
+	protocol     string
+	address      string
+	externaladdr string
+
+	inbound  pipemap
+	outbound pipemap
+
+	peerlist *types.Peerlist
+
+	listener struct {
+		sync.Mutex
+		*net.TCPListener
+	}
+	listening atomic.Bool
+
+	done    chan uint
+	ready   chan uint
+	closed  chan uint
+	errored chan uint
+
+	sink     chan types.Work
+	handlers map[types.PeerEvent][]func(...interface{})
+
+	logger struct {
+		sync.RWMutex
+		print func(...interface{}) (int, error)
+	}
+
+	err struct {
+		sync.Mutex
+		error
+	}
+}
+
+var (
+	ErrNotCreated = errors.New("Module error: P2P Module not created. Trying to start the p2p module before calling create.")
+)
+
+func NewP2PModule() *P2PModule {
+	return &P2PModule{
+		protocol: Protocol,
+		address:  Address,
+
+		inbound:  *NewIoMap(MaxInbound),
+		outbound: *NewIoMap(MaxOutbound),
+
+		c: NewDomainCodec(),
+
+		peerlist: nil,
+		sink:     make(chan types.Work, 100), // TODO: rethink whether this should be buffered
+
+		done:   make(chan uint, 1),
+		ready:  make(chan uint, 1),
+		closed: make(chan uint, 1),
+
+		handlers: make(map[types.PeerEvent][]func(...interface{}), 0),
+		errored:  make(chan uint, 1),
+	}
+}
+
+func Create(config *config.Config) (modules.NetworkModule, error) {
+	module := NewP2PModule()
+
+	module.setLogger(func(args ...interface{}) (int, error) {
```

Comment:
My preference: remove custom logger logic. This is something that all interfaces will need and we'll probably add as a single "infra" PR. 

I think @iajrz has been thinking about this.

Reply:
Agreed, this is a placeholder as some other parts of this code. I expect that PR then to remove this.